### ISSUE TITLE
On the performance

### ISSUE THREAD

**Issue by donproc** [Tai-Hsien/MeshSegNet#33]
Hi, @Tai-Hsien
We met one performance issue. 

Training set: ~800
Val set:         ~100
Number of cells: 6000
Random sampling
Epochs: 200
LR:1e-3
Optimizer: ADAM
Batch size: 10

The validation DSC and SEN are always below 0.9. And we can not get result ~0.98 in paper.
Do you have any suggestions?

**Comment by Tai-Hsien** [comment]
Are the numbers with post-processing (i.e., graph cut)?

**Comment by donproc** [comment]
> 

No, we did not count post-processing in. The DSC and SEN in validation during training is around 0.9.

**Comment by Tai-Hsien** [comment]
Post-processing is quite important if you would like to achieve a good number.
In addition, what's the number of cells for the downsampled scan? Our downsampled mesh has roughly 10,000 cells, and 6000 cells selected from 10,000 cells during each iteration.

**Comment by donproc** [comment]
> Post-processing is quite important if you would like to achieve a good number. In addition, what's the number of cells for the downsampled scan? Our downsampled mesh has roughly 10,000 cells, and 6000 cells selected from 10,000 cells during each iteration.

@Tai-Hsien Sorry for reply so late.
We select 6000 cells from original mesh(~150000 cells, depends on source of mesh) directly during training. 
We just down sampling mesh to 10000 cells as your script in inference.

**Comment by brennenhuang** [comment]
share the code I implement for downsampling (150000cells get 10000cells)
`
from sklearn.neighbors import KNeighborsClassifier
import vedo
mesh = vedo.load(path)
neigh = KNeighborsClassifier(n_neighbors=3)
neigh.fit(mesh.cellCenters(), np.ravel(mesh.getCellArray('Label')))

#downsample to 10000
target_num = 10000
ratio = target_num/mesh.NCells() # calculate ratio
mesh2 = mesh.clone()
mesh2.decimate(fraction=ratio)

fine_labels = neigh.predict(mesh2.cellCenters())
fine_labels = fine_labels.reshape(-1, 1)
mesh.addCellArray(fine_labels, 'Label')

`
